biojava-structure/src/main/java/org/biojava/nbio/structure/io/mmcif/SimpleMMcifConsumer.java

```diff
@@ -328,7 +328,13 @@ public void newAtomSite(AtomSite atom) {
 
 		String recordName    = atom.getGroup_PDB();
 		String residueNumberS = atom.getAuth_seq_id();
-		Integer residueNrInt = Integer.parseInt(residueNumberS);
+		Integer residueNrInt;
+		if(residueNumberS != null) {
+			residueNrInt = Integer.parseInt(residueNumberS);
+		} else {
+			String label_seq_id = atom.getLabel_seq_id();
+			residueNrInt = Integer.parseInt(label_seq_id);
```

Comment:
`label_seq_id` is not the same as residue number. This can introduce many problems.

Reply:
We should discuss this further.

I added it because the atp.cif.gz file I added (from pymol) has label_seq_id but not auth_seq_id. It seems reasonable to me to fall back on the sequential numbering if the authors don't specify a custom numbering.

Note that `auth_seq_id` is optional but `label_seq_id` is required. From the spec it sounds to me like `label_seq_id` is a good fallback for ResidueNumber:

[_atom_site.auth_seq_id:](http://mmcif.wwpdb.org/dictionaries/mmcif_pdbx_v50.dic/Items/_atom_site.auth_seq_id.html)
> An alternative identifier for _atom_site.label_seq_id that
> may be provided by an author in order to match the identification
> used in the publication that describes the structure.

Ideally BioJava would use `Group.getId()` rather then `getResidueNumber()` everywhere. However, many things still use residue numbers, so setting a default seems prudent.

Reply:
There's another potential issue here, which is that the seq_id is stored as a `long` while ResidueNumber contains an `Integer`. I don't think there are any files with >2billion groups per entity, but if we hit it then the ResidueNumber would throw a NumberFormatException here. I think that's fine.

Reply:
I started #775 to continue this discussion.